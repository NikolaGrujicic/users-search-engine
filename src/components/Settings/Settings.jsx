import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { getRandomUsersAction, getRandomUsersByNationalityAction } from '../../redux/actions/userActions';

const Settings = () => {
  const dispatch = useDispatch();

  const allLocal = localStorage.getItem('all');
  const gbLocal = localStorage.getItem('gb');
  const esLocal = localStorage.getItem('es');
  const chLocal = localStorage.getItem('ch');
  const frLocal = localStorage.getItem('fr');

  const gb = gbLocal === 'true' ? 'gb' : '';
  const es = esLocal === 'true' ? 'es' : '';
  const ch = chLocal === 'true' ? 'ch' : '';
  const fr = frLocal === 'true' ? 'fr' : '';

  const [allState, setAllState] = useState(allLocal === 'true');
  const [gbState, setGbState] = useState(gbLocal === 'true');
  const [esState, setEsState] = useState(esLocal === 'true');
  const [chState, setChState] = useState(chLocal === 'true');
  const [frState, setFrState] = useState(frLocal === 'true');

  const onSubmit = e => {
    e.preventDefault();
    if (allState) {
      dispatch(getRandomUsersAction());
    }
    dispatch(getRandomUsersByNationalityAction(gb, es, ch, fr));
  };
  return (
    <div className="container settings">
      <div className="row">
        <span className="setting-heading">Filter users by nationality:</span>
        <br />
        <form onSubmit={onSubmit}>
          <div className="form-group">
            <div className="input-group-text setting-checkbox-container">
              All Nationalities
              <input
                name="all"
                onChange={e => {
                  setAllState(e.target.checked);
                  localStorage.setItem(e.target.name, e.target.checked);
                }}
                checked={allState}
                className="setting-checkbox-input"
                type="checkbox"
                aria-label="Checkbox for following text input"
              />
            </div>
          </div>
          <div className="form-group">
            <div className="input-group-text setting-checkbox-container">
              United Kingdom
              <input
                name="gb"
                onChange={e => {
                  setGbState(e.target.checked);
                  localStorage.setItem(e.target.name, e.target.checked);
                }}
                checked={gbState}
                className="setting-checkbox-input"
                type="checkbox"
                aria-label="Checkbox for following text input"
              />
            </div>
          </div>
          <div className="form-group">
            <div className="input-group-text setting-checkbox-container">
              Spain
              <input
                name="es"
                onChange={e => {
                  setEsState(e.target.checked);
                  localStorage.setItem(e.target.name, e.target.checked);
                }}
                checked={esState}
                className="setting-checkbox-input"
                type="checkbox"
                aria-label="Checkbox for following text input"
              />
            </div>
          </div>
          <div className="form-group">
            <div className="input-group-text setting-checkbox-container">
              Switzerland
              <input
                name="ch"
                onChange={e => {
                  setChState(e.target.checked);
                  localStorage.setItem(e.target.name, e.target.checked);
                }}
                checked={chState}
                className="setting-checkbox-input"
                type="checkbox"
                aria-label="Checkbox for following text input"
              />
            </div>
          </div>
          <div className="form-group">
            <div className="input-group-text setting-checkbox-container">
              France
              <input
                name="fr"
                onChange={e => {
                  setFrState(e.target.checked);
                  localStorage.setItem(e.target.name, e.target.checked);
                }}
                checked={frState}
                className="setting-checkbox-input"
                type="checkbox"
                aria-label="Checkbox for following text input"
              />
            </div>
          </div>
          <button type="submit" className="btn btn-primary">Filter</button>
        </form>
      </div>
    </div>
  );
};

export default Settings;
