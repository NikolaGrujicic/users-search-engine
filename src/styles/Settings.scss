#settings {
    margin-top: 100px;
    max-width: 1240px;
}

.settings-filter {
    width: 300px;
}

.setting-heading {
    width: 100%;
    font-size: 20px;
}

.setting-checkbox-container {
    padding: 10px;
    margin-bottom: 10px;
}

.setting-checkbox-input {
    margin-left: 5px;
}