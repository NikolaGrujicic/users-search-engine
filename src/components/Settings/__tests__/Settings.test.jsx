import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import Settings from '../Settings';

const middlewares = [thunk];

const mockStore = configureStore(middlewares);

const initialState = {
  randomUsers: {
    users: [],
    loading: false,
    scrolling: false,
  },
};

let store;
test('renders Settings', () => {
  store = mockStore(initialState);
  const wrapper = shallow(
    <Provider store={store}>
      <Settings />
    </Provider>,
  );
  expect(wrapper).toMatchSnapshot();
});
