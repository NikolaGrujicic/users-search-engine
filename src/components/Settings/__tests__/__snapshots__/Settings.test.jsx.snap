// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container settings"
>
  <div
    className="row"
  >
    <span
      className="setting-heading"
    >
      Filter users by nationality:
    </span>
    <br />
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <div
          className="input-group-text setting-checkbox-container"
        >
          All Nationalities
          <input
            aria-label="Checkbox for following text input"
            checked={false}
            className="setting-checkbox-input"
            name="all"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="input-group-text setting-checkbox-container"
        >
          United Kingdom
          <input
            aria-label="Checkbox for following text input"
            checked={false}
            className="setting-checkbox-input"
            name="gb"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="input-group-text setting-checkbox-container"
        >
          Spain
          <input
            aria-label="Checkbox for following text input"
            checked={false}
            className="setting-checkbox-input"
            name="es"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="input-group-text setting-checkbox-container"
        >
          Switzerland
          <input
            aria-label="Checkbox for following text input"
            checked={false}
            className="setting-checkbox-input"
            name="ch"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="input-group-text setting-checkbox-container"
        >
          France
          <input
            aria-label="Checkbox for following text input"
            checked={false}
            className="setting-checkbox-input"
            name="fr"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Filter
      </button>
    </form>
  </div>
</div>
`;
